### DISCUSSION TITLE
Rename `Control` to `NodeUI`

### DISCUSSION THREAD

**Discussion by AaronRecord** [godotengine/godot-proposals#4969] [GUI]
I think this change would better match the current naming scheme, then we'd have `Node2D` for 2D nodes, `Node3D` nodes, and `NodeUI` for UI nodes.

~~I'm not familiar with any other engine or framework that uses the term `Control` to describe a UI component, but  there's also plenty of engine's and frameworks I'm not familiar with so~~ I'd be happy to hear arguments otherwise.

**Comment by theraot** [comment]
The name `NodeUI` sounds good to me. However, something similar to what you say happens: who else call them that? I would expect people to search for ["Widget"](https://en.wikipedia.org/wiki/Graphical_widget) (window gadget) or… "Control".

> I'm not familiar with any other engine or framework that uses the term `Control` to describe a UI component

Microsoft.

Microsoft Visual Basic 1.0 is as far back as I can trace the term, both the DOS and the Windows 3.0 version use the term "Control". Windows API uses the term, see [Windows Controls](https://docs.microsoft.com/en-us/windows/win32/api/_controls/). COM kept the term, see [ActiveX Controls](https://docs.microsoft.com/en-us/windows/win32/com/activex-controls). And .NET still uses it, see [Control class](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.control).

They are, of course, components. Which is the approach that replaced Model-View-Controller (think Smalltalk not Ruby on Rails). If I were to speculate… Well, enough said.

**Comment by Calinou** [comment]
This was already proposed in https://github.com/godotengine/godot-proposals/issues/763 a while ago, and it was rejected.
